### ISSUE TITLE
Add Fairseq NMT as the Backend-Friendly Model Variant

### ISSUE THREAD

**Issue by mhs4670go** [Samsung/TICO#124]
## Goal  

Seed the new folder structure with a **sequence-to-sequence NMT model** built on *fairseq* and tuned for the TRIV backend. This serves as a worked example of the workflow proposed in the parent issue #95.

## Target Model

| Item                | Value |
|---------------------|-------|
| Upstream framework  | [fairseq](https://github.com/facebookresearch/fairseq) ([ecbf110](https://github.com/facebookresearch/fairseq/commit/ecbf110e1eb43861214b05fa001eff584954f65a)) |
| Model        | `transformer.wmt16.en-de` (encoder 6-layers / decoder 6-layers) |

## Expected Folder Layout

```bash
models/
└── fairseq/
    ├── baseline_fp32/
    │   └── model.py
    └── TRIV/
        ├── model.py     
        ├── modules/
        │   ├── transformer_encoder_triv.py
        │   └── transformer_decoder_triv.py
        ├── modeling_fairseq_triv.py # (optional) whole-file override
        ├── config.json
        └── README.md
```
* Only the modified classes live under `modules/`.  
* If >50 % of `fairseq/models/transformer.py` needs editing, copy it to `modeling_fairseq_triv.py` and document that choice in the README.

## Acceptance Criteria

* ✅ Model exports to Circle **without unsupported op fallback**.
* ✅ Compiles and runs on TRIV device.
* ✅ CI green on both driver-level and per-module unit tests.  

Once merged, I hope this Fairseq NMT variant will become the template for other seq-to-seq and multilingual models.

**Comment by mhs4670go** [comment]
## (Encoder Layer) About input shape

To export `TransformerEncoderBase` for hardware that doesn’t support dynamic shapes, we can use a fixed-length dummy input and a corresponding padding mask. For example, if our maximum sequence length is 64 and batch size is 1:

```python
import torch
import fairseq

import tico

# 1) Prepare the model
en2de = torch.hub.load('pytorch/fairseq', 'transformer.wmt16.en-de',
                       tokenizer='moses', bpe='subword_nmt')
en2de.eval()  # disable dropout
model = en2de.models[0]

# 2) Create dummy inputs
batch_size = 1
max_seq_len = 64
embed_dim = en2de.cfg.model.encoder_embed_dim

dummy_x = torch.randn(max_seq_len, batch_size, embed_dim)
dummy_padding_mask = torch.zeros(batch_size, max_seq_len, dtype=torch.bool)

# 3) Export the model
cm = tico.convert(model.encoder.layers[0], (dummy_x, dummy_padding_mask))
cm.save("encoder_layer.circle")
```

### Key points

- Fixing seq_len=64 avoids dynamic-shape requirements.
- Passing the `dummy_padding_mask` ensures any padding logic continues to work correctly.
- Adjust `max_seq_len` to the longest sequence we need to support.

**Comment by mhs4670go** [comment]
![Image](https://github.com/user-attachments/assets/3918c2dd-d64b-4e21-94ba-0b39d9392ec9)

**Comment by mhs4670go** [comment]
> Why the reshape should remain?

I'm not sure because this is from torch's export behavior. But, this can be removed with #126.
